src/main/java/software/amazon/awssdk/crt/mqtt/MqttClientConnection.java

```diff
@@ -49,14 +52,45 @@ void deliver(String topic, byte[] payload, boolean dup, int qos, boolean retain)
         }
     }
 
+    /**
+     * Static help function to create a MqttConnectionConfig from a
+     * Mqtt5ClientOptions
+     */
+    private static MqttConnectionConfig s_toMqtt3ConnectionConfig(Mqtt5ClientOptions mqtt5options) throws Exception {
+        try {
+            MqttConnectionConfig options = new MqttConnectionConfig();
+            options.setEndpoint(mqtt5options.getHostName());
+            options.setPort(Math.toIntExact(mqtt5options.getPort() != null ? mqtt5options.getPort() : 0));
+            options.setSocketOptions(mqtt5options.getSocketOptions());
+            if (mqtt5options.getConnectOptions() != null) {
+                options.setClientId(mqtt5options.getConnectOptions().getClientId());
+                options.setKeepAliveSecs(
+                        Math.toIntExact(mqtt5options.getConnectOptions().getKeepAliveIntervalSeconds() != null
+                                ? mqtt5options.getConnectOptions().getKeepAliveIntervalSeconds()
+                                : 0));
+            }
+            options.setCleanSession(
+                    mqtt5options.getSessionBehavior().compareTo(Mqtt5ClientOptions.ClientSessionBehavior.CLEAN) <= 0);
+            options.setPingTimeoutMs(
+                    Math.toIntExact(mqtt5options.getPingTimeoutMs() != null ? mqtt5options.getPingTimeoutMs() : 0));
+            options.setProtocolOperationTimeoutMs(Math.toIntExact(
+                    mqtt5options.getAckTimeoutSeconds() != null ? mqtt5options.getAckTimeoutSeconds() : 0) * 1000);
+            return options;
+        } catch (ArithmeticException e) {
```

Comment:
Still not seeing the value in rethrowing an exception that wraps the same exception type.